api/models/app.go

```diff
@@ -474,18 +486,22 @@ func (a *App) RunAttached(process, command, releaseId string, height, width int,
 		return err
 	}
 
-	// If no releaseId is provided, use the last promoted release to run this process
-	// otherwise iterate over the previous revisions (starting with the latest) looking for the releaseId specified.
-	if releaseId == "" {
-		releaseId = a.Release
-
-		for _, container = range task.TaskDefinition.ContainerDefinitions {
-			if *container.Name == process {
-				break
-			}
+	for _, container = range task.TaskDefinition.ContainerDefinitions {
+		if *container.Name == process {
+			break
 		}
+	}
 
-	} else {
+	if container == nil {
+		return fmt.Errorf("unable to find container for %s", process)
+	}
+
+	// If the release ID provided does not equal the active one, some logic is needed to determine the next steps.
+	// - For a previous release, we iterate over the previous TaskDefinition revisions (starting with the latest) looking for the releaseId specified.
+	// - If the release has yet to be promoted, we use the most recent TaskDefinition with the provided release's environment.
+	if releaseId != a.Release {
+
+		var releaseContainer *ecs.ContainerDefinition
```

Comment:
This can indeed take a while, especially if a user calls `convox run` with a very old release. The go-ecs API doesn't provide a "paging" functionality like it's other calls for describing task definitions (which is understandable since we want to describe one revision).

Two options top of mind:

1. Sleep for a second after _n_ number of task definitions. (arbitrarily start with 15?).
2. Limit `convox run` to use only one of the last _n_ task definitions. (again 15 might work).

These options help with rate limits. There's still the issue of a run taking too long. Especially if it's towards the bottom of the list. More recent releases would be towards the top of the list and are iterated over first.

I'm leaning towards option 2. As is, to determine if a release hasn't been promoted, we have to iterate the task definitions revisions. If it's not found there then it's an "unpromoted" release. It would be a bad user experience having to wait that iteration if there are hundreds of revisions to iterate first.